.github/workflows/main.yml

```diff
@@ -46,22 +46,45 @@ jobs:
             git submodule update --init --recursive
             make html
 
-    # Deployment to GH Pages
-  deploy:
+  # Builds the WebPages locally with Publish settings before Deployment
+  publish:
     # Runner Type(s)
     runs-on: ubuntu-latest
-    # Dependents on Build 
-    needs: build
     strategy:
       matrix:
         python-version: [3.9]
 
+    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      - name: Displaying Python Versions
-        run: python -c "import sys; print(sys.version)"
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      # Set up Python Environment based on the version
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      # Installations for all the dependencies 
+      - name: Install dependencies
+        run: |
+            python -m pip install --upgrade pip
+            pip install -r requirements.in
+      # Add the support plugins required & builds the webpages in HTML format
+      - name: Build the webpages
+        run: |
+            git submodule update --init --recursive
+            make publish
 
-    # # Uncomment for deployment
-    # steps:
-    # - name: Deployment to Github Pages 
-    #   run: |
-    #       make publish-to-github
\ No newline at end of file
+  # Deployment to GH Pages
+  deploy:
+    # Runner Type(s)
+    runs-on: ubuntu-latest
+    # Dependents on Publish
+    needs: publish
+    strategy:
+      matrix:
+        python-version: [3.9]
+
+    steps:
+      - name: Deployment to GitHub Pages
+        run: make publish-to-github
```

Comment:
Doesn't this need some sort of token?

Reply:
Yes indeed ... It will need some token for deployment from your side ... but using the `make` commands I think so the tokens can be configured in the configuration file but first I will make the above changes, then will try to check how to go about it :)

Reply:
👍🏽

Reply:
@astrojuanlu I found this awesome extension https://github.com/marketplace/actions/github-pages-pelican-build-action ... I have configured it but its need the owner's rights to make it work as seen here :
![image](https://user-images.githubusercontent.com/66117751/117431311-b853c000-af46-11eb-94b9-1d0da3d23d5f.png)
Can you please pull this branch & check for these changes?

Reply:
It says:

> ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'

The error is elsewhere.

Reply:
It was showing something to do with the ownership & maintainers rights ... Its awesome we were able to fix it :)

Reply:
I had a question @astrojuanlu, by using `make publish-to-github-force` how are we bypassing the prompt for the github id & password that is required to publish the site to gh pages?

Reply:
I too was thinking about this but didn't implement it as I was not absolutely sure that it would work ... its awesome that it works :)

Reply:
Got it :) ... Thanks @astrojuanlu